client/src/app/views/admin/components/data-upload-modal/data-upload-modal.component.ts

```diff
@@ -25,33 +30,62 @@ export class DataUploadModalComponent implements OnInit {
-
-    if (!dataFileInput.files || !dataFileInput.files[0]) {
-      this.toastService.toast("Datový soubor je povinný.", "notice");
+  async uploadData(form: NgForm, dataFileInput: HTMLInputElement, eventsFileInput: HTMLInputElement, accountingFileInput: HTMLInputElement, paymentsFileInput: HTMLInputElement) {
+    const fields: [HTMLInputElement, string, string, string][] = [
+      [dataFileInput, "data", "dataFileMode", "Datový soubor"],
+      [eventsFileInput, "events", "eventsFileMode", "Číselník"],
+      [accountingFileInput, "accounting", "accountingFileMode", "Rozpočet"],
+      [paymentsFileInput, "payments", "paymentsFileMode", "Faktury"]
+    ]
+    if (!this.year) {
+      this.toastService.toast("Nezvolený rok", "notice") 
       return;
     }
-    if (!this.year) return;
-
-    const formData = form.value;
-    const data = new FormData();
-
-    data.set("year", String(this.year));
-
-    data.set("validity", formData.validity);
-
-    const file = dataFileInput.files[0]
-    data.set("dataFile", file, file.name);
-
-    if (eventsFileInput.files && eventsFileInput.files[0]) {
-      const file = eventsFileInput.files[0]
-      data.set("eventsFile", file, file.name);
+    if (!fields.map(a => a[0]).some((e: HTMLInputElement) => e.files && e.files[0])) {
+      this.toastService.toast("Žádné soubory nevybrány k nahrání", "notice") 
+      return;
     }
 
-    await this.importService.importAccounting(this.profileId, data);
+    const v = form.value
+    const tasks: Promise<String>[] = []
+
+    // Using for...of here so return actually returns (and doesn't just break, like it does in forEach)
+    for (const [element, name, mode, error] of fields) {
+      const data = new FormData()
+      if (element.files && element.files[0]) {
+        if (v[mode] == "") {
+          this.toastService.toast(`${error} nemá nastavený režim nahrání.`, "notice")
+          return
+        }
+        const file = element.files[0]
+        data.set(name, file, file.name)
+        data.set("year", String(this.year));
+        data.set("validity", v.validity);
+        
+        switch(name) {
+          case "payments":
+            tasks.push(this.importService.importPayments(this.profileId, data, v[mode] == "append"))
+            break
+          case "accounting":
+            tasks.push(this.importService.importAccounting(this.profileId, data, v[mode] == "append"))
+            break
+          case "events":
+            tasks.push(this.importService.importEvents(this.profileId, data, v[mode] == "append"))
+            break
+          case "data":
+            tasks.push(this.importService.importData(this.profileId, data, v[mode] == "append"))
+            break
+        }
+      } 
+    }
 
+    await Promise.all(tasks).then(_ => {
+      this.toastService.toast("Data nahrána pro import.", "notice")
+    }).catch(e => {
+      this.toastService.toast("Neočekáváná chyba při odesílání dat. Prosím, kontaktujte administrátora.", "notice")
```

Comment:
All the toasts in this component are of the type "notice". Consider using "error" type where appropriate